### PR TITLE
Issue 7993 documentation for gift registry mutations

### LINKED ISSUE #7993: GraphQL: Add gift registry mutations
# Community engineering issue

Type of Doc:

-  [x] New topic
-  [ ] Topic Update, Devdoc link:
-  [ ] Other

Magento 2 GitHub Issue: https://github.com/magento/magento2/issues/28586
Magento 2 GitHub PR: https://github.com/magento/partners-magento2ee/pull/280/

Magento 2 version:

-  [x] 2.4.2

Notifications of Changes:

-  [ ] Changes to Admin interface?
-  [ ] New software requirements?
-  [x] New REST endpoints or updated GraphQL query syntax?
-  [ ] New command line option?

## Content

Add documentation for the following gift registry mutations

createGiftRegistry
shareGiftRegistry

Also update the customer object to include gift registry objects and attributes

### PR DESCRIPTION
**Pull request by dineshvb**
## Purpose of this pull request

This pull request (PR) fixes #7993 

## Affected DevDocs pages

/graphql/mutations/create-gift-registry.html
/graphql/mutations/share-gift-registry.html

## Links to Magento Source Code

magento/partners-magento2ee#280

### CODE REVIEW

**Comment by dineshvb** [comment]
@dobooth There seems to be a GITHUB_TOKEN missing error. Can you see through it?

**Comment by dobooth** [comment]
Yup, I saw that. Something amiss on our end.

**Comment by jeff-matthews** [comment]
This is more likely an issue with the Superlinter. It's safe to ignore since it's not a required check and will not block merging.

**Comment by keharper** [comment]
Note that gift registry is a Commerce feature, not B2B. In the TOC, change `edition: b2b-only` to `edition: ee-only`. Also change `b2b` to `ee` in the frontmatter of the other two topics.

**Comment by keharper** [comment]
Also @dineshvb , can you make sure you've pulled in the latest changes into the `2.4.2-develop` branch? There's way too many unrelated changes in this PR.

File: src/_data/toc/graphql.yml [hunk 1]
```diff
@@ -265,6 +265,11 @@ pages:
           edition: ee-only
           exclude_versions: ["2.3"]
 
+        - label: createGiftRegistry mutation
+          url: /graphql/mutations/create-gift-registry.html
+          edition: b2b-only
```
Comment (rogyar) [inline src/_data/toc/graphql.yml:270]: Please, correct me if I'm wrong, but this feature is available for ee as well

```suggestion
          edition: ee-only
```

File: src/_data/toc/graphql.yml [hunk 2]
```diff
@@ -430,6 +435,11 @@ pages:
           edition: ee-only
           exclude_versions: ["2.3"]
 
+        - label: shareGiftRegistry mutation
+            url: /graphql/mutations/share-gift-registry.html
+            edition: b2b-only
```
Comment (rogyar) [inline src/_data/toc/graphql.yml:440]:
```suggestion
            edition: ee-only
```

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+## Input attributes
+
+The `CreateGiftRegistryInput` input object defines the gift registry.
+
+### CreateGiftRegistryInput attributes {#CreateGiftRegistryInput}
+
+The `CreateGiftRegistryInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/create-gift-registry.md:191]: Please sort the attributes alphabetically by the attribute name.

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+### AddGiftRegistryRegistrantInput attributes {#AddGiftRegistryRegistrantInput}
+
+The `AddGiftRegistryRegistrantInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`first_name` | String! | First name of the registrant
+`last_name` | String! | Last name of the registrant
+`email` | String! | Email id of the registrant
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/create-gift-registry.md:211]:
```suggestion
`email` | String! | Email address of the registrant
```

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`gift_registry` | GiftRegistry | Contains the created gift registry
+
+### GiftRegistry attributes {#GiftRegistry}
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`id`| ID! |Id of the created gift registry
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/create-gift-registry.md:235]:
```suggestion
`id`| ID! |ID of the created gift registry
```

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+`dynamic_attributes` | [GiftRegistryDynamicAttribute] | Certain additional attributes are specified as a code-value pair
+`items` | [GiftRegistryItemInterface] | The items in the gift registry
+
+### GiftRegistryRegistrant attributes {#GiftRegistryRegistrant}
+
+The `GiftRegistryRegistrant` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`id` | ID! | Id of the registrant
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/create-gift-registry.md:254]:
```suggestion
`id` | ID! | ID of the registrant
```

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+`first_name` | String! | First name of the registrant
+`last_name` | String! | Last name of the registrant
+`email` | String! | Email id of the registrant
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/create-gift-registry.md:257]:
```suggestion
`email` | String! | Email address of the registrant
```

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+`label` | String! | A label for the additional attribute
+`value` | String! | A corresponding value for the code
+
+### GiftRegistryItemInterface attributes {#GiftRegistryItemInterface}
+
+The `GiftRegistryItemInterface` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`id` | String! | Id of the item
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/create-gift-registry.md:285]:
```suggestion
`id` | String! | ID of the item
```

File: src/guides/v2.4/graphql/mutations/share-gift-registry.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+     }
+  }
+}
+```
+
+## Input attributes
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`id`| ID! | Id of the gift registry to be shared
```
Comment (rogyar) [inline src/guides/v2.4/graphql/mutations/share-gift-registry.md:76]:
```suggestion
`id`| ID! | ID of the gift registry to be shared
```

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @dineshvb. Thank you for your collaboration. Good job! 
Please check my suggestions below.

**Comment by keharper** [comment]
@dineshvb I will be processing #8172 first. Take a look at the comments I left there and be ready to replace the description of the `GiftRegistry` object with a link to an include file, if it's the same object returned by the `giftRegistry` query.

**Comment by dineshvb** [comment]
@rogyar I will do the changes and give the PR in a few hours

File: src/guides/v2.4/graphql/mutations/create-gift-registry.md (hunk 3 above)
Comment (dineshvb) [reply src/guides/v2.4/graphql/mutations/create-gift-registry.md:191]: I have sorted it

**Review by rogyar — APPROVED** [review]

File: src/guides/v2.4/graphql/mutations/share-gift-registry.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
+---
+group: graphql
+title: shareGiftRegistry mutation
+ee_only: true
+contributor_name: Zilker Technology
+contributor_link: https://www.ztech.io/
+---
+The `shareGiftRegistry` mutation shares a gift registry for the logged in customer.
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/share-gift-registry.md:8]:
```suggestion
The `shareGiftRegistry` mutation allows the logged in customer to share a gift registry.
```

File: src/guides/v2.4/graphql/mutations/share-gift-registry.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  }
+}
+```
+
+## Input attributes
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`id`| ID! | ID of the gift registry to be shared
+`invitees`| [ShareGiftRegistryInviteeInput!]! | Invitees' details
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/share-gift-registry.md:77]:
```suggestion
`invitees`| [ShareGiftRegistryInviteeInput!]! | A list of people invited to participate in the event
```

File: src/guides/v2.4/graphql/mutations/share-gift-registry.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+`sender`| ShareGiftRegistrySenderInput! | Sender's details
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/share-gift-registry.md:78]:
```suggestion
`sender`| ShareGiftRegistrySenderInput! | Information about the invitation sender
```

File: src/guides/v2.4/graphql/mutations/share-gift-registry.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+### ShareGiftRegistrySenderInput attributes {#ShareGiftRegistrySenderInput}
+
+The `ShareGiftRegistrySenderInput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`message` | String! | A brief message from the sender
+`name`| String! | Name of the sender
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/share-gift-registry.md:87]:
```suggestion
`name`| String! | The name of the sender
```

File: src/guides/v2.4/graphql/mutations/share-gift-registry.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+Attribute |  Data Type | Description
+--- | --- | ---
+`message` | String! | A brief message for the invitee
+`name`| String! | Name of the invitee
+
+## Output attributes
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`is_shared` | Boolean! | Conveys whether the gift registry has been shared/not
```
Comment (keharper) [inline src/guides/v2.4/graphql/mutations/share-gift-registry.md:102]:
```suggestion
`is_shared` | Boolean! | Indicates whether the gift registry was successfully shared
```

**Review by keharper — CHANGES_REQUESTED** [review]

**Comment by keharper** [comment]
@dineshvb Rebase this PR to `gift-registry` and make the updates I requested.

**Review by keharper — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**